src/hotspot/share/oops/klass.inline.hpp

```diff
@@ -36,6 +36,12 @@ inline oop Klass::klass_holder() const {
   return class_loader_data()->holder();
 }
 
+inline void Klass::keep_alive() const {
+  // Resolving the holder (a WeakHandle) will keep the klass alive until the next safepoint.
+  // Making the klass's CLD handle oops (e.g. the java_mirror), safe to store in the object graph.
+  static_cast<void>(klass_holder());
+}
+
```

Comment:
> safe to store in the object graph.

It could be good to also mention roots like Handles, since that seems to be a common misunderstanding:

> safe to store in the object graph and it's roots (E.g. Handles).

Reply:
Why is not java_mirror() extended to set this keep_alive() of the CLD oop-stuff under the hood?

Reply:
> Why is not java_mirror() extended to set this keep_alive() of the CLD oop-stuff under the hood?

Would like to improve our API to have the CLD oop handles accessors work more like WeakHandles w.r.t. peek and resolve. But would need to go over all the CLD oops resolves, because a few of them should be peeks. 

Side note: OopHandle::peek should probably be removed now that single generational ZGC is gone, as it has the same semantics as OopHandle::resolve.

> Can you add a comment to Klass::java_mirror() that says it doesn't keep the loader alive, "see above".

Added a comment to Klass::java_mirror() and updated the Klass::keep_alive() comment.

Reply:
Yes, I think we're going to need to relearn how to access oops.